src/DonorDashboards/Tabs/EditProfileTab/AvatarRoute.php

```diff
@@ -49,10 +50,23 @@ public function handleRequest(WP_REST_Request $request)
 
         // Delete existing Donor profile avatar attachment
         if (give()->donorDashboard->getAvatarId()) {
+            if (!$this->avatarBelongsToCurrentUser()) {
```

Comment:
Would this make more sense as a `permission_check` on the route registration?

Reply:
It turns out all these donor dashboard routes share the same `permission_check` which is all about making sure the donor is valid and logged in.  I could have overloaded that here but figured it would be better to address the single scope of this issue that was reported.  It would be great to clean up these routes in the future so they can have a top level permission check but have the ability to extend that permission per route.